riscv/triggers.h

```diff
@@ -208,6 +208,7 @@ class mcontrol_common_t : public trigger_t {
 
 protected:
   match_t legalize_match(reg_t val) const noexcept;
+  bool legalize_timing(reg_t val, reg_t timing_mask, reg_t select_mask, reg_t execute_mask, reg_t load_mask) const noexcept;
```

Comment:
This could be `static` now. (And same for `legalize_match()` while you're here.)

`legalize_action()` has a similar issue to your previous implementation of `legalize_timing()` because it looks at `dmode`.

Reply:
The gcc complained that the "static member function cannot have cv-qualifier," which means the "static" cannot be with "const noexcept." I prefer keeping the cv-qualifiers ("const noexcept") because we don't need to call the functions with no object.

Referene: https://stackoverflow.com/questions/19812786/why-cant-a-static-member-function-have-a-cv-qualifier

Reply:
Commit https://github.com/riscv-software-src/riscv-isa-sim/pull/1214/commits/798e0daf667ccff0f9bc8d68ad6bf91d0c351dad removes legalize_action()'s dependency on ```dmode```.

Reply:
@scottj97 Should we keep the "const noexcept"?? Or, we can move ```legalize_timing()``` and ```legalize_match()``` out of the classes and make them available in triggers.cc only.

Reply:
`const` has no meaning (I guess?) when it's a static method. (Although what about static attributes? Hmm. But this class has no such attributes, so we don't care.)

_cv-qualifier_ does not include `noexcept`, so you should be able to keep that.

So, better to make it `static` and remove `const`.